to-do-list/src/app.ts

```diff
@@ -0,0 +1,91 @@
+import express, { Request, Response } from "express";
+import cors from "cors";
+import { AddressInfo } from "net";
+import connection from "./connection";
+
+const app = express();
+
+app.use(express.json());
+app.use(cors());
+
+// 1- Criar usuário
+
+const createUser = async (
+  id: number,
+  name: string,
+  nickname: string,
+  email: string
+) => {
+  await connection
+    .insert({
+      id: id,
+      name: name,
+      nickname: nickname,
+      email: email
+    })
+    .into("Users")
+}
+
+app.post('/user', async (req: Request, res: Response) => {
+  try {
+    const { name, nickname, email } = req.body
+
+    if (!name && !nickname && !email) {
+      throw new Error("Please, fill in the fiels!")
+    }
+
+    await createUser(
+      req.body.id,
+      req.body.name,
+      req.body.nickname,
+      req.body.email
+    )
```

Comment:
Aqui vc já havia criado variáveis com essas propriedades do body na linha 31, não precisa se referir ao req.body novamente. Só uma questão de estilo de código.
Também, o id está sendo obtido pelo body, mas não se lança erro se ele não for passado na requisição (no if acima, linha 33). Então, se o id não for passado, essa função executa com undefined.